ansible/roles/data-products-deploy/templates/submit-script.j2

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+export SPARK_HOME={{ analytics.home }}/spark-{{ spark_version }}-bin-hadoop2.7
+export MODELS_HOME={{ analytics.home }}/models-{{ model_version }}
+export DP_LOGS={{ analytics.home }}/logs/data-products
+export KAFKA_HOME={{ analytics.home }}/kafka_2.11-0.10.1.0
+
+## Job to run daily
+cd /mount/data/analytics/scripts
+source model-config.sh
+today=$(date "+%Y-%m-%d")
+
+## job broker-list and kafka-topic
+zookeeper={{ zookeeper }}
+brokerList={{ brokerlist }}
+job_topic={{ analytics_job_queue_topic }}
+unique_topic={{ env }}.telemetry.unique
+
+while :; do
+    case $1 in
+        -c|--job)   shift 
+                    job="$1"            
+        ;;
+        -m|--mode)  shift 
+                    type="$1"            
+        ;;
+        -p|--partitions)    shift 
+                            partitions=$1          
+        ;;
+        -s|--parallelisation)   shift 
+                                parallelisation=$1            
+        ;;
+        -sd|--startDate)   shift 
+                            start_date=$1            
+        ;;
+        -ed|--endDate)     shift 
+                            end_date=$1            
+        ;;
+        -h|--sparkMaster)   shift 
+                            sparkMaster=$1            
+        ;;
+        *) break
+    esac
+    shift
+done
+
+if [ -z "$sparkMaster" ]; then sparkMaster="local[*]"; else sparkMaster="$sparkMaster"; fi
+
+if [ "$type" = "via-partition" ]; then
+    # get partitions from kafka metadata
+    if [ -z "$partitions" ]; then partitions=`$KAFKA_HOME/bin/kafka-topics.sh --zookeeper $zookeeper --describe --topic $unique_topic   | awk 'NR==1{print $2}' |  awk -F":" '{print $2}'`; else partitions=$partitions; fi
+    endPartitions=`expr $partitions - 1`
+    if [ -z "$parallelisation" ]; then parallelisation=1; else parallelisation=$parallelisation; fi
+    # add partitions to config and start jobs
+    for i in $(seq 0 $parallelisation $endPartitions)
+        do 
+            # add partitions to config
+            partitionString="\"delta\":0,\"partitions\":[$(seq -s , $i `expr $i + $parallelisation - 1`)]"
+            if [ -z "$start_date" ]; then
+                job_config=$(config $job)
+                finalConfig=${job_config/'"delta":0'/$partitionString}
+                echo $finalConfig
+                echo "Running $job by partitions."
+                nohup $SPARK_HOME/bin/spark-submit --master $sparkMaster --jars $(echo ${libs_path}/lib/*.jar | tr ' ' ','),$MODELS_HOME/analytics-framework-2.0.jar,$MODELS_HOME/scruid_2.11-2.3.2.jar,$MODELS_HOME/batch-models-2.0.jar --class org.ekstep.analytics.job.JobExecutor $MODELS_HOME/batch-models-2.0.jar --model "$job" --config "$finalConfig" >> "$DP_LOGS/$today-job-execution.log" 2>&1
```

Comment:
I think we do not need a nohup when using spark-submit. We can still redirect the output to a log file.

Reply:
Updated spark-submit without nohup